graphs/graphs.go

```diff
@@ -52,42 +52,29 @@ func New(input string) (*Graph, error) {
 }
 
 func readNodes(r io.Reader) (map[uint64]*Node, error) {
-	input := make([]byte, 8)
-	for {
-		_, err := r.Read(input)
-		if err == io.EOF {
-			break
-		}
-	}
-
+	scanner := bufio.NewScanner(r)
+	scanner.Split(bufio.ScanLines)
 	nodes := make(map[uint64]*Node)
 	var current *Node
-	for _, c := range string(input) {
-		switch {
-		case unicode.IsDigit(c):
-			id, err := strconv.ParseUint(string(c), 10, 64)
-			if err != nil {
-				return nil, err
-			}
 
-			if current == nil {
-				current = &Node{id: id}
-			} else {
-				previous := current
-				current = &Node{id: id}
-				previous.edges = append(previous.edges, current)
-			}
-
-			nodes[id] = current
-		// Space character except newline or EOF -> ignore.
-		case unicode.IsSpace(c) && c != '\n' || c == 0:
-			continue
-		// Newline -> create a new node on next iteration.
-		case c == '\n':
-			current = nil
-		default:
+	for scanner.Scan() {
+		ids := strings.Fields(scanner.Text())
+		nodeID, err := strconv.ParseUint(ids[0], 0, 64)
```

Comment:
what will happen if line is empty? ids[0]

Reply:
ParseUint(ids[0], 0, 64)

although it works, you are using base 10, it isn't clear as is and relies checking implementation

ParseUint(ids[0], 10, 64)

Reply:
> what will happen if line is empty? ids[0]

`scanner.Scan` returns `false` on that case and the loop isn't executed. There's a test for that case.

Reply:
> ParseUint(ids[0], 0, 64)
> 
> although it works, you are using base 10, it isn't clear as is and relies checking implementation
> 
> ParseUint(ids[0], 10, 64)

Good point, it does make it clearer! Thanks!

Reply:
> 
> 
> > what will happen if line is empty? ids[0]
> 
> `scanner.Scan` returns `false` on that case and the loop isn't executed. There's a test for that case.

it is fail if it contains a \n for example: https://play.golang.org/p/EXzWZystbFe

I guess it is outside project aim, more just fyi